stdlib/public/runtime/ImageInspectionStatic.cpp

```diff
@@ -14,22 +14,24 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include "ImageInspection.h"
-#include <cstring>
-
 // Currently only tested on linux but should work for any ELF platform
 #if defined(__ELF__) && defined(__linux__)
 
+#include "ImageInspection.h"
+#include <cstring>
+
 // These are defined in swift_sections.S to mark the start of a section with the
 // length of the data followed immediately by the section data
 struct alignas(uint64_t) Section;
 extern const Section protocolConformancesStart asm(".swift2_protocol_conformances_start");
 extern const Section typeMetadataStart asm(".swift2_type_metadata_start");
 
-struct SectionInfo {
-  uint64_t size;
-  const char *data;
-};
+using namespace swift;
+
+// Called from ImageInspectionInit
+void
+swift::addNewDSOImage(const void *addr) {
+}
```

Comment:
This shouldn't have to exist at all in a static executable. I'd rather we get a link error if we accidentally try to link in the static constructor stub than quietly link in a useless static constructor.